ambry-router/src/test/java/com/github/ambry/router/NonBlockingRouterTest.java

```diff
@@ -715,6 +715,87 @@ public void onBlobDeleted(String blobId, String serviceId, Account account, Cont
     Assert.assertEquals("All operations should have completed", 0, router.getOperationsCount());
   }
 
+  /**
+   * Test that even when a composite blob put succeeds, the slipped put data chunks are deleted.
+   */
+  @Test
+  public void testSuccessfulPutDataChunkDelete() throws Exception {
```

Comment:
We should monitor how this test case performs on CI runs. If it fails intermittently, we may have to make some changes.

I understand why it is probabilistic though.